cmd/metricscollector/v1beta1/file-metricscollector/main.go

```diff
@@ -179,78 +185,78 @@ func watchMetricsFile(mFile string, stopRules stopRulesFlag, filters []string) {
 		// Print log line
 		klog.Info(logText)
 
-		// Check if log line contains metric from stop rules.
-		isRuleLine := false
-		for _, rule := range stopRules {
-			if strings.Contains(logText, rule.Name) {
-				isRuleLine = true
-				break
-			}
-		}
-		// If log line doesn't contain appropriate metric, continue track file.
-		if !isRuleLine {
-			continue
-		}
-
-		// If log line contains appropriate metric, find all submatches from metric filters.
-		for _, metricReg := range metricRegList {
-			matchStrings := metricReg.FindAllStringSubmatch(logText, -1)
-			for _, subMatchList := range matchStrings {
-				if len(subMatchList) < 3 {
-					continue
-				}
-				// Submatch must have metric name and float value
-				metricName := strings.TrimSpace(subMatchList[1])
-				metricValue, err := strconv.ParseFloat(strings.TrimSpace(subMatchList[2]), 64)
-				if err != nil {
-					klog.Fatalf("Unable to parse value %v to float for metric %v", metricValue, metricName)
+		switch *metricsFileFormat {
+		case commonv1beta1.TextFormat.String():
```

Comment:
Can we convert the metricsFileFormat to `FileSystemFileFormat` then match it here?

Reply:
Thanks for your comments! @gaocegege 

Does it mean to declare `type metricFileFormatFlag FileSystemFileFormat` like [`stopRulesFlag`](https://github.com/kubeflow/katib/blob/2a0b12eba755e4a47fa9d96e3102f31d5a25eebe/cmd/metricscollector/v1beta1/file-metricscollector/main.go#L62)?

Reply:
I think we can use a string var flag, then convert the string to FileSystemFileFormat in the code. Because `case xxx.String()` looks strange WDYT

```
format := FileSystemFileFormat(metricsFileFormat)
switch format {
  case commonv1beta1.TextFormat:
}
```

Reply:
It sounds good to me!
I will modify all `commonv1beta1.TextFormat.String()` and `commonv1beta1.JsonFormat.String()` in `cmd/metricscollector/v1beta1/file-metricscollector` and `pkg/metricscollector/v1beta1/file-metricscollector` packages.

Reply:
Done.